bundles/org.openhab.core.karaf/src/main/java/org/openhab/core/karaf/internal/jaas/ManagedUserRealm.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.karaf.internal.jaas;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Singleton;
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.AppConfigurationEntry.LoginModuleControlFlag;
+import javax.security.auth.spi.LoginModule;
+
+import org.apache.karaf.jaas.boot.ProxyLoginModule;
+import org.apache.karaf.jaas.config.JaasRealm;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.openhab.core.auth.UserRegistry;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A JAAS realm description for the {@link UserRegistry} based login module.
+ *
+ * @author Yannick Schaus - initial contribution
+ */
+@Singleton
+@Component(service = JaasRealm.class)
+@Service
+public class ManagedUserRealm implements JaasRealm {
+
+    public static final String REALM_NAME = "openhab";
+    public static final String MODULE_CLASS = "org.openhab.core.auth.jaas.internal.ManagedUserLoginModule";
+
+    private final Logger logger = LoggerFactory.getLogger(ManagedUserRealm.class);
+
+    BundleContext bundleContext;
+
+    @Activate
+    public ManagedUserRealm(BundleContext bundleContext, @Reference LoginModule loginModule) {
+        this.bundleContext = bundleContext;
+        logger.info("Using login module {} for the openhab realm", loginModule.getClass().getCanonicalName());
+    }
+
+    @Override
+    public String getName() {
+        return REALM_NAME;
+    }
+
+    @Override
+    public int getRank() {
+        return 1;
+    }
+
+    @Override
+    public AppConfigurationEntry[] getEntries() {
+        Map<String, Object> options = new HashMap<>();
+        options.put(ProxyLoginModule.PROPERTY_MODULE, MODULE_CLASS);
+        // options.put(ProxyLoginModule.PROPERTY_BUNDLE, bundleContext.getBundle().getBundleId());
```

Comment:
can this be removed?

Reply:
I wasn't sure if it was needed, but it seems to do fine without it. And it's not correct anyway because the login module is in the `org.openhab.core.auth.jaas` bundle, not this one. I'll remove it.